### PR TITLE
componentsbuild

### PR DESCRIPTION
**Pull request by Tyrabims**

### CODE REVIEW

File: src/Forms.module.css [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  justify-content: space-between;
+} */
+
+
+/* @media screen and (min-width: 500px) {
+  .containerr{
+    display:inline-block;
+    justify-content: center;
+  }
+} */
```
Comment (greatertomi) [inline src/Forms.module.css]: Remove this comment

File: src/components/Navbar.tsx [hunk 2]
```diff
@@ -1,16 +1,25 @@
-import React from 'react';
-import { Link } from "react-router-dom";
+import { NavLink } from 'react-router-dom';
+// import logo from '../assets/logo.webp';
```
Comment (greatertomi) [inline src/components/Navbar.tsx]: Remove this

File: src/index.css [hunk 3]
```diff
@@ -1,52 +1,21 @@
...
-    color: #213547;
-    background-color: #ffffff;
-  }
-  a:hover {
-    color: #747bff;
-  }
-  button {
-    background-color: #f9f9f9;
-  }
-} */
```
Comment (greatertomi) [inline src/index.css]: Remove this

File: src/components/Navbar.tsx [hunk 4]
```diff
@@ -1,16 +1,24 @@
-import React from 'react';
```
Comment (greatertomi) [inline src/components/Navbar.tsx:1]: Why did you delete this? You need to import `React` to use jsx

File: src/components/Navbar.tsx [hunk 5]
```diff
@@ -1,16 +1,24 @@
...
-import { Link } from "react-router-dom";
+import { NavLink } from 'react-router-dom';
+import { Navbar as NavbarBs, Container, Nav } from 'react-bootstrap';
+import '../index.css';
+
 
 const Navbar = () => {
```
Comment (greatertomi) [inline src/components/Navbar.tsx]: Eslint throwing `ESLint: Unexpected block statement surrounding arrow body; move the returned value immediately after the `=>`.(arrow-body-style)`

Fix it.

File: src/components/UserDetailsProps.tsx [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        const { data } = await apiClient.get(`/users/${username}/repos?per_page=5`);
+        setUserRepos(data);
+      } catch {
+        throw new Error('Error fetching data');
+      }
+    };
+    fetchData();
+    fetchUserRepos();
+  }, []);
+  console.log(userRepos);
```
Comment (greatertomi) [inline src/components/UserDetailsProps.tsx]: Delete this

File: src/components/UserDetailsProps.tsx [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+      }
+    };
+    const fetchUserRepos = async () => {
+      try {
+        const { data } = await apiClient.get(`/users/${username}/repos?per_page=5`);
+        setUserRepos(data);
+      } catch {
+        throw new Error('Error fetching data');
+      }
+    };
```
Comment (greatertomi) [inline src/components/UserDetailsProps.tsx]: These two functions do the same thing. Group them together like this and delete one
```suggestion
    const fetchData = async () => {
      try {
        const { data: userData } = await apiClient.get(`/users/${username}`);
        const { data: userReposData } = await apiClient.get(`/users/${username}/repos?per_page=5`);
        setUserDetails(userData);
        setUserRepos(userReposData);
      } catch {
        throw new Error('Error fetching data');
      }
    };
```

File: src/components/UserDetailsProps.tsx [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+      </Button>
+      Hireable: {userDetails?.hireable ? <FaCheck /> : <BsFillXCircleFill />}
+      <Card className="w-85 mt-4">
+        <Card.Body>
+          <Card.Text>
+            <Row>
+              <Col className="text-center">
+                <Image
+                  className=" rounded-circle mb-3"
+                  style={{ width: 150 }}
```
Comment (greatertomi) [inline src/components/UserDetailsProps.tsx]: Remove this inline style

File: src/components/UserDetailsProps.tsx [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+            Public Repos: {userDetails?.public_repos}
+          </Button>
+          <Button variant="dark" className="d-inline me-2 btn-sm">
+            Public Gists: {userDetails?.public_gists}
+          </Button>
+        </Card.Body>
+      </Card>
+      <ListGroup className="w-85 mt-3 text-danger">
+        {userRepos.map((userRepo) => (
+          <ListGroup.Item className="mb-3">{userRepo?.name}</ListGroup.Item>
```
Comment (greatertomi) [inline src/components/UserDetailsProps.tsx]: Add a key to this map

File: src/index.css [hunk 10]
```diff
@@ -1,52 +1,21 @@
...
-  text-decoration: none;
+body:hover{
+  background-color: aliceblue;
 }
 
-li h1 {
-  padding-right: 1000px;
+/* form */
+.search{
+  background-color: none;
```
Comment (greatertomi) [inline src/index.css]: background-color does not have `none` property. You can remove it.

File: src/index.css [hunk 11]
```diff
@@ -1,52 +1,21 @@
...
-    color: #747bff;
-  }
-  button {
-    background-color: #f9f9f9;
-  }
-} */
+@media screen and (max-width: 700px) {
+   .search{
+    padding:20px 150px;
+   }
```
Comment (greatertomi) [inline src/index.css]: Format this properly. And also, there are so many many unnecessary spacing in this file. Trim that too

File: src/pages/About.tsx [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+
+export const About = () => {
+  return (
+  <div>
+    <h2>About This App</h2>
+    <h5>App to search Github users<br/>
+    Version 1.0.0
+    </h5>
+  </div>)
+}
```
Comment (greatertomi) [inline src/pages/About.tsx]: - Format this file properly
- Fix all eslint error including the `arrow-body-style` error.

File: src/pages/Home.tsx [hunk 13]
```diff
@@ -0,0 +1,9 @@
+import Form from '../components/Form';
```
Comment (greatertomi) [inline src/pages/Home.tsx:2]: Import React here

File: src/pages/Home.tsx [hunk 14]
```diff
@@ -0,0 +1,9 @@
...
+
+export const Home = () => {
```
Comment (greatertomi) [inline src/pages/Home.tsx]: Fix Eslint `arrow-body-style` error here.

File: src/services/api-client.ts [hunk 15]
```diff
@@ -0,0 +1,6 @@
+import axios from 'axios';
+
+
+export default axios.create({
+  baseURL:"https://api.github.com"
+})
```
Comment (greatertomi) [inline src/services/api-client.ts]: Fix all the eslint errors in this file.

**Review by greatertomi — CHANGES_REQUESTED** [review]
Generally, it looks like you don't give a damn about eslint. Is your IDE configured to see eslint promptings?

File: src/components/UserDetailsProps.tsx (hunk 6 above)
Comment (Tyrabims) [reply src/components/UserDetailsProps.tsx]: okay sir!

File: src/components/Navbar.tsx (hunk 4 above)
Comment (Tyrabims) [reply src/components/Navbar.tsx:1]: fixed!

File: src/components/Navbar.tsx (hunk 5 above)
Comment (Tyrabims) [reply src/components/Navbar.tsx]: Okay

File: src/components/UserDetailsProps.tsx (hunk 8 above)
Comment (Tyrabims) [reply src/components/UserDetailsProps.tsx]: okay done

File: src/index.css (hunk 10 above)
Comment (Tyrabims) [reply src/index.css]: okay

File: src/pages/About.tsx (hunk 12 above)
Comment (Tyrabims) [reply src/pages/About.tsx]: okay done

File: src/pages/Home.tsx (hunk 13 above)
Comment (Tyrabims) [reply src/pages/Home.tsx:2]: okay done

File: src/pages/Home.tsx (hunk 14 above)
Comment (Tyrabims) [reply src/pages/Home.tsx]: okay done

File: src/components/UserDetailsProps.tsx (hunk 6 above)
Comment (greatertomi) [reply src/components/UserDetailsProps.tsx]: You forgot this.

**Review by greatertomi — APPROVED** [review]
